db/migrate/20170822145654_add_user_id_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserIdToPosts < ActiveRecord::Migration[5.1]
+  def change
+    add_column :posts, :user_id, :integer
```

Comment:
https://stackoverflow.com/questions/22815009/add-a-reference-column-migration-in-rails-4

When you already have users and uploads tables and wish to add a new relationship between them.

All you need to do is: just generate a migration using the following command:

```ruby
rails g migration AddUserToUploads user:references
```
Which will create a migration file as:
```ruby
class AddUserToUploads < ActiveRecord::Migration
  def change
    add_reference :uploads, :user, index: true
  end
end
```
Then, run the migration using rake db:migrate. This migration will take care of adding a new column named user_id to uploads table (referencing id column in users table), PLUS it will also add an index on the new column.